### DISCUSSION TITLE
Image labels annotation enhancement

### DISCUSSION THREAD

**Discussion by EmilienAlvarez** [biigle/core#589] [Ideas]
Hello,

I use Biigle.de for annotating image labels, but it seems that the application is mainly dedicated to annotating polygons. I think a few simple improvements could make the job of annotating image labels easier, for example:

- in the label tree of the full-format image viewer, it might be interesting to have a direct way (one or two clicks) of assigning one or more labels.
- Largo's re-evaluation could include annotation of image labels.
- Charts generation too

Thanks,
 Emilien

**Comment by mzur** [comment]
Hi, thank you for the feedback!

> in the label tree of the full-format image viewer, it might be interesting to have a direct way (one or two clicks) of assigning one or more labels.

You can select up to 10 labels as "favourites" by clicking on the star icon of the label in the label trees tab of the image annotation tool. These labels can be selected with the shortcut keys 0-9. If you now open the image labels tab in the annotation tool, you can select a label quickly using the shortcut keys and then click on the button to attach the label. Does this help?

> Largo's re-evaluation could include annotation of image labels.

Can you please describe a specific workflow? The volume overview can work similar to Largo in this regard. You can apply a filter to show only images having a specific label and then update the image labels there. Are the user interactions (applying filter, selecting label, deleting/attaching label) too complicated compared to Largo?

> Charts generation too

This is an excellent idea. I've put it on our roadmap (#591). Which chart types do you think make sense for file labels?

**Comment by EmilienAlvarez** [comment]
Hello,

Thanks for your tips!

Keyboard shortcuts 0-9 can indeed help save time but I still think that a small '+' button that would appear in front of each label (like the favorite star) to add the label image could be very convenient and more suitable for a large tree of labels.

Indeed, using user interactions as filters is very practical, simple and can work in a similar way to Largo, thank you.

I suppose the same graphs as annotated polygons would be interesting: annotated vs. non-annotated, label distribution, label co-occurrence.

**Reply by mzur** [reply]
> I still think that a small '+' button that would appear in front of each label (like the favorite star) to add the label image could be very convenient and more suitable for a large tree of labels.

What about a new search input field in the image labels tab where you can type a label name (similar to the input in the label trees tab)? While you type you get label suggestions in a dropdown. Once you hit <kbd>Enter</kbd> on a suggestion, the label is attached to the image. Would this also work for you? It might be even faster than clicking on buttons.

**Reply by EmilienAlvarez** [reply]
Yes, indeed, I think it would be faster !

**Reply by mzur** [reply]
Great! This is now tracked in https://github.com/biigle/core/issues/594.
